fall-2020/navigation.py

```diff
@@ -70,24 +66,117 @@ def set_desired_position(self, desired_position):
+            #coordinate value taken at one go and then seperated as inputting the coordinates seperately will take longer time
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+        #error checking for if the x given coordinate  is already behind the machine or not
+        if self.current_position(0) > self.desired_position(0): 
+            print("desired position for x axis is behind the TBM ,please put x coordinate which is infront of TBM \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+            
+        targetreached = False
+        while targetreached != True :
+            #should add a pausing mechanism to let actuators take input and implement movement before checking new position and distance left to move using time.sleep(.) 
+            #confused about whether the steering methods return true if full movement to desired location complete or only if movement started and movement distance is given
+            #if steering methods return true after their full movement completed ,then loop is not required .
+            self.update_current_position()
+            #differences between the coordinates is calculated
+            difference_x = self.desired_position(0) - self.current_position(0)
+            difference_y = self.desired_position(1) - self.current_position(1)
+            difference_z = self.desired_position(2) - self.current_position(2)
+
+            #checking each axis for which direction to move:front,right and above are taken to be positive while their opposite directions are taken as negative 
+            #for the x axis
+            if difference_x == 0:
+                pass
+            else:
+                statusx = self.steering.move_forward()
+                if statusx == False:
+                    print("error for forward moving actuator")
+                    return False
+                #shouldn't the move_forward method also take the distance parameter?so that it hows how much to move forward
```

Comment:
It could, or for moving forwards and stopping you could actually poll to see if you have reached the desired x coordinate yet. But if you would like to change the steering methods to take a distance parameter, for consistency with the other methods, I have no issue with that

Reply:
For now, I think leaving the steering method as it is, for now, will be better.